context.go

```diff
@@ -0,0 +1,44 @@
+package pq
+
+import "time"
+
+// contextInterface and background are copied from the context package.
+
+// contextInterface is named this way so that files tagged go1.8 can import
+// context without conflict.
+type contextInterface interface {
+	Deadline() (deadline time.Time, ok bool)
+	Done() <-chan struct{}
+	Err() error
+	Value(key interface{}) interface{}
+}
+
+type emptyCtx int
```

Comment:
You are correct, but in order to be a more direct copy of the context code I kept it as an int: https://tip.golang.org/pkg/context/?m=all#emptyCtx